### ISSUE TITLE
Categories cannot been marked Read automatically

### ISSUE THREAD

**Issue by R-J** [vanilla/vanilla#6856]
The CommentModel->setWatch method is used for determining if a category is shown as "Read" or not. That method contains the following lines:
~~~
$dateMarkedRead = val('DateMarkedRead', $category);
if ($dateMarkedRead) {
~~~
As a result only categories which have been marked as read manually can be shown as read.

I think it should be handled like that:
~~~
$wheres = ['CategoryID' => $categoryID];
if (val('DateMarkedRead', $category)) {
    $wheres['DateLastComment>'] = $dateMarkedRead;
}
$discussions = $discussionModel->get(0, $lookBackCount + 1, $wheres);
~~~

I will do a PR if https://github.com/vanilla/vanilla/pull/6855 has been approved or declined.

**Comment by charrondev** [comment]
- Related https://github.com/vanilla/vanilla/issues/4013
- Related https://github.com/vanilla/vanilla/issues/2357

**Comment by theknigge** [comment]
@charrondev do you have an idea when you can fix that?

**Comment by charrondev** [comment]
@R-J had taken a pretty good swing at it with https://github.com/vanilla/vanilla/pull/6859 but seems to have made some mistake in how he branched the 2 PRs. I tried to reconcile his PR but ultimately could not fix the conflicts in a reasonable amount of time.

Personally it's not on our immediate roadmap, although I understand the annoyance. We are accepting contributions though. @R-J's would have been merged (and mostly fixed the issue) if the conflicts could have been resolved.
